sympy/polys/ring_series.py

```diff
@@ -572,6 +574,108 @@ def _coefficient_t(p, t):
             p1[monomial_div(expv, expv1)] = p[expv]
     return p1
 
+def rs_series_reversion_newton(p,x,n):
+
+    r"""
+    Fast series reversion (aka Lagrange Inversion) using Newton-like updates.
+
+    Given a ring polynomial $p$ in variable $x$, this method computes a ring polynomial
+    $q$ in $x$ such that $q(p(x)) = x + O(x^n)$.
+
+    The series should be of form $p(x) = a_1 x + a_2 x^2 + ...$ where $a_1$ is invertible.
+
+    Parameters
+    ==========
+
+    p : :class:`~.PolyElement` to be inverted.
+    x : :class:`~.PolyElement` with respect to which ``p`` is inverted.
+    n : number of expansion terms to be computed.
+
+    Note
+    ====
+
+    The algorithm adapts the Newton method to solve a general equation
+    $$
+    f(x)=y.
+    $$
+    By Taylor's expansion:
+    $$
+    f(x+h) = f(x) + f'(x) h + O(h^2).
+    $$
+    If $f(x) = y + O(y^{k+1})$ then with $h = -\frac{f(x)-y}{f'(x)} = O(y^{k+1})$ we obtain:
+    $$
+    f(x+h)-y = O(h^2) = O(y^{2k+2}).
+    $$
+    Thus, updating $x\gets x-h$ increases the accuracy from $k$ to $2k+1$ terms.
+
+    Examples
+    ========
+
+    The example below shows how to invert the expresion $f=\mathrm{e}^{x}-1$ around $x=0$:
+
+    >>> from sympy.polys.domains import QQ
+    >>> from sympy.polys.rings import ring
+    >>> from sympy.polys.ring_series import rs_exp, rs_series_reversion_newton
+    >>> Rx, xr = ring('x', QQ)
+    >>> n = 4
+    >>> fr = rs_exp(xr,xr,n)-1
+    >>> rs_series_reversion_newton(fr,xr,n)
+    1/3*x**3 - 1/2*x**2 + x
+
+    Another example below shows general formulas for few first coefficients of the inversion expansion:
```

Comment:
This and other lines are too long and should be wrapped.

Reply:
What is too long? Where it is defined? Perhaps should be verified with `black`.